src/components/Recorder/index.test.js

```diff
@@ -1,79 +1,178 @@
-        stop: jest.fn(), 
-        ondataavailable: jest.fn(),
-        onstop: jest.fn()
-    })
-    const div = document.createElement('div');
-    ReactDOM.render(<Recorder />, div);
-})
+const mockUseMediaRecorder = (defaultRecordings = []) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const [isRecording, setIsRecording] = useState(false)
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const [recordings, setRecordings] = useState(defaultRecordings)
 
-it('is a snapshot', () => {
-    useMediaRecorder.mockReturnValue({state: 'inactive', start: jest.fn(), stop: jest.fn(), ondataavailable: jest.fn()})
-    const component = renderer.create(
-        <Recorder />,
-    );
-    let tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
-})
+  const recorder = {
+    start: () => setIsRecording(true),
+    stop: () => {
+      setRecordings(currentList => [...currentList, createMockRecording({ idNumber: currentList.length })])
+      setIsRecording(false)
+    },
+  };
 
-describe('Recording audio', () => {
-    global.URL.createObjectURL = jest.fn();
-    it('record turns colors when clicked', async () => {
-        // has recording-audio been added to the classes?
-        useMediaRecorder.mockReturnValue({state: 'inactive', start: jest.fn(), stop: jest.fn(), ondataavailable: jest.fn()})
-        render(<Recorder />)
-        const recordButton = await screen.findByRole('button')
-        expect(recordButton).toBeInTheDocument()
-        expect(recordButton).toHaveTextContent('Record')
-        userEvent.click(recordButton)
-        await screen.findByText('Stop')
-        expect(recordButton).toHaveClass('recording-audio')
-    })
-})
+  return {recorder, recordings, setRecordings, isRecording}
+};
+
+describe('With an empty list of recordings', () => {
+  beforeEach(() => {
+    useMediaRecorder.mockImplementation(() => mockUseMediaRecorder());
+    render(<Recorder stream={{}}/>);
+  });
+
+  it('renders without crashing', () => {
+    const button = screen.getByRole("button", {name: 'Record'});
+    expect(button).toBeInTheDocument();
+    expect(button).toHaveClass('record-play')
+  });
 
+  it('user can start a recording pressing the button', () => {
+    const button = screen.getByRole("button", { name: 'Record' })
+    expect(button).toHaveClass('record-play')
+    fireEvent.click(button)
+    expect(button).toHaveTextContent(/stop/i);
+  });
 
-it('recording can be renamed', async () => {
-    const newName = 'The new name'
-    global.prompt = () => newName // https://stackoverflow.com/questions/41732903/stubbing-window-functions-in-jest
+  it('record button turns red while recording', () => {
+    const button = screen.getByRole("button", { name: /record/i });
+    expect(button).toHaveClass('record-play')
+    fireEvent.click(button)
+    expect(button).toHaveClass('record-play', 'recording-audio')
+  })
 
+  it('adds a new recording to the list when the user clicks stop', () => {
```

Comment:
This one… What exactly are we testing here? It seems to me like we're testing the functionality of the mocked hook rather than the hook itself. Is this a valid test and if so, why?